signer/src/config.rs

```diff
@@ -333,13 +333,28 @@ where
     Ok(v)
 }
 
+/// Format an error message for an invalid private key length.
+fn format_invalid_private_key_length_msg(len: usize) -> String {
+    format!(
+        "Invalid Stacks private key length (got {}, expected 64 or 66)",
+        len
+    )
+}
+
 /// A deserializer for the [`PrivateKey`] type. Returns an error if the private
 /// key is not valid hex or is not the correct length.
 fn private_key_deserializer<'de, D>(deserializer: D) -> Result<PrivateKey, D::Error>
 where
     D: Deserializer<'de>,
 {
-    PrivateKey::from_str(&String::deserialize(deserializer)?).map_err(serde::de::Error::custom)
+    let s = String::deserialize(deserializer)?;
+    if [64, 65].contains(&s.len()) {
+        PrivateKey::from_str(&s[..64]).map_err(serde::de::Error::custom)
+    } else {
+        Err(serde::de::Error::custom(
+            format_invalid_private_key_length_msg(s.len()),
```

Comment:
Hmmm, I see. You use this function so that you can more easily match on the message in the tests. And you match on the message since this is a user facing error. Also the existing error variant for this, `InvalidPrivateKeyLength`, isn't quite right.

Well, I think it would be better to control error messages from a single place (or at least a single kind of place), which would be the error variant display message. Maybe create new error variants?

Reply:
Fixed in [d179173](https://github.com/stacks-network/sbtc/pull/485/commits/d1791732d24772a4b7ad07ce2a12e286005b70b8)

Tbh I don't think it matters so much when dealing with config error messages since they're generally one-off, i.e. we already use `ConfigError::Message(...)` in this file a fair bit. But now I've added a variant, so we can migrate those messages to the enum as well in a later PR.

Reply:
👍🏽 oh right, good point.